autogpt_platform/backend/migrations/20250205100104_add_profile_trigger/migration.sql

```diff
@@ -0,0 +1,108 @@
+CREATE OR REPLACE FUNCTION add_user_and_profile_to_platform()
+RETURNS TRIGGER AS $$
+DECLARE
+    -- Random username generation
+    selected_adjective TEXT;
+    selected_animal    TEXT;
+    random_int         INT;
+    generated_username TEXT;
+BEGIN
+    -- Exit early if NEW.id is null to prevent constraint violations
+    IF NEW.id IS NULL THEN
+        RAISE EXCEPTION 'Cannot create user/profile: id is null';
+    END IF;
+
+    /*
+      1) Insert into platform."User"
+         (If you already have such a row or want different columns, adjust below.)
+    */
+    INSERT INTO platform."User" (id, email, "updatedAt")
+    VALUES (NEW.id, NEW.email, now());
+
+    /*
+      2) Generate a random username and verify uniqueness
+         -- This generates random combinations until finding an unused username
+         -- Will raise an exception if unable to generate a unique username after 10 attempts
+    */
+    FOR i IN 1..10 LOOP
+        SELECT unnest
+          INTO selected_adjective
+          FROM (VALUES ('happy'), ('clever'), ('swift'), ('bright'), ('wise'), ('funny'), ('cool'), ('awesome'), ('amazing'), ('fantastic'), ('wonderful')) AS t(unnest)
+          ORDER BY random()
+          LIMIT 1;
+
+        SELECT unnest
+          INTO selected_animal
+          FROM (VALUES ('fox'), ('wolf'), ('bear'), ('eagle'), ('owl'), ('tiger'), ('lion'), ('elephant'), ('giraffe'), ('zebra')) AS t(unnest)
+          ORDER BY random()
+          LIMIT 1;
+
+        SELECT floor(random() * (99999 - 10000 + 1) + 10000)::int
+          INTO random_int;
+
+        generated_username := lower(selected_adjective || '-' || selected_animal || '_' || random_int);
+
+        -- Check if username is already taken
+        IF NOT EXISTS (
+            SELECT 1 FROM platform."Profile" WHERE username = generated_username
+        ) THEN
+            -- Username is unique, exit the loop
+            EXIT;
+        END IF;
+
+        -- If we've tried 10 times and still haven't found a unique username
+        IF i = 10 THEN
+            RAISE EXCEPTION 'Unable to generate unique username after 10 attempts';
```

Comment:
Do we really need to have a clever name here? can we randomize it and make this query easier to understand?
Is this for all users or only for cornercases ? won't the user need to be asked to fill these detail during the signup ?

Reply:
No we don't. However, I felt it looked nicer than displaying a hash, currently we don't have a flow where they are asked to enter their username during signup.

Reply:
I've moved it too a new function